Framework/API/inc/MantidAPI/WorkspaceProperty.tcc

```diff
@@ -182,7 +190,7 @@ std::string WorkspaceProperty<TYPE>::setDataItem(
   boost::shared_ptr<TYPE> typed = boost::dynamic_pointer_cast<TYPE>(value);
   if (typed) {
     std::string wsName = typed->getName();
```

Comment:
I don't think `wsName` is now used anywhere else other than the assign below so we could just do
```
m_workspaceName = typed->getName()
```
to avoid the temporary copy?